org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/model/ModelViewer.java

```diff
@@ -235,30 +260,24 @@ public void keyPressed(KeyEvent e) {
 				}
 			}
 		});
-		
-		if (PageUtil.getPage()!=null) {
-			ISelection selection = PageUtil.getPage().getSelection();
-			processWorkbenchSelection(selection); // If model view is selected later but something it can process is the page selection...
-			
-			if (model==null) { // Go and look for the model on a view
-				final IViewPart part  = PageUtil.getPage().findView(ScanView.ID);
-				final Object    model = part!=null ? part.getAdapter(IScanPathModel.class) : null;
-				processObject(model);
-			}
-		}
-		
-		createActions();
-		
-		return (U)content;
+
+		return viewer;
 	}
 
 	private void createActions() {
 		
-		List<IContributionManager> mans = null;
+		if (site==null) return;
 		
+		List<IContributionManager> mans = Arrays.asList(site.getActionBars().getToolBarManager(), site.getActionBars().getMenuManager());
+		
+		// TODO We should be able to switch around different roles
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)